lib/inc/internal/facts/resolvers/operating_system_resolver.hpp

```diff
@@ -216,6 +216,20 @@ namespace facter { namespace facts { namespace resolvers {
          * @return Returns the resolver data.
          */
         virtual data collect_data(collection& facts);
+
+        /**
+         * Collects the resolver's kernel data.
+         * @param facts The fact collection that is resolving facts.
+         * @param result The current resolver data.
+         */
+        virtual void collect_kernel_data(collection& facts, data &result);
```

Comment:
I was trying to find examples but it's still unclear to me, @branan or @MikaelSmith do we want auxiliary methods like these to be private like in https://github.com/puppetlabs/facter/blob/master/lib/inc/internal/facts/resolvers/networking_resolver.hpp or just protected like in https://github.com/puppetlabs/facter/blob/master/lib/inc/internal/facts/resolvers/virtualization_resolver.hpp? I feel like these could easily be private given how they're used.